mountpoint-s3-client/tests/head_bucket.rs

```diff
@@ -41,3 +41,16 @@ async fn test_head_bucket_forbidden() {
         Err(ObjectClientError::ServiceError(HeadBucketError::PermissionDenied(_)))
     ));
 }
+
+#[tokio::test]
+async fn test_head_bucket_not_found() {
+    let client = get_test_client();
+    let bucket = "nosuch..bucket"; // Invalid bucket name, so cannot be created by anyone.
```

Comment:
Think I'd rather use a bucket name that is explicitly documented to not exist, like `DOC-EXAMPLE-BUCKET`, rather than inventing our own.

Reply:
I thought the same. The issue is that these buckets actually exist - they are just owned by AWS to prevent them being grabbed by others.

Reply:
The uppercase versions don't exist, and so this is actually also (surprisingly) a good test of whether we're preserving case in bucket names or not.

Reply:
Ah, interesting. I've updated the test and it does pass using that casing. It switches over to path-based addressing, so it is case sensitive with this behavior.

I'll push an update soon.